---
title: 'Figure 1: PerIS in homeostasis'
author: "Andrew Y.F. Li Yim"
date: '2023-07-26'
output: html_document
---

```{r libraries, include=FALSE}
library(Seurat)
library(ggplot2)
library(dplyr)
library(ggrastr)
library(ggrepel)
library(ggpubr)
library(ggalt)
library(viridis)
library(readr)
```

If the snakemake pipeline was run as-is, the following files will be generated in the folder from which the pipeline was run. If necessary, adjust paths accordingly.

## Setup

### Paths

```{r paths}
base_path <- "/mnt/smb/ltytgat/ayliyim/MI-group/ayliyim/projects/TIPRJ0000015_MAPS/maps_crc" # L-drive

# Subsets
seurat_pf_rds <- file.path(base_path, "output/q1_pf_characterization/subsets/hc_pf_SeuratObject.Rds")
seurat_pf_myeloid_rds <- file.path(base_path, "output/q1_pf_characterization/subsets/hc_pf_Myeloid_SeuratObject.Rds")
seurat_pf_macrophages_rds <- file.path(base_path, "output/q1_pf_characterization/subsets/hc_pf_macrophages_SeuratObject.Rds")

seurat_pbmc_pf_liver_colon_rds <- file.path(base_path, "output/q1_pf_characterization/subsets/pbmc_pf_liver_colon_SeuratObject.Rds")
seurat_liver_macrophages_rds <- file.path(base_path, "output/q1_pf_characterization/subsets/liver_macrophages_SeuratObject.Rds")
seurat_colon_macrophages_rds <- file.path(base_path, "output/q1_pf_characterization/subsets/colon_macrophages_SeuratObject.Rds")

tam_markers_xlsx <- file.path(base_path, "config/genes_of_interest/tams_ma_2022.xlsx")
pf_macrophages_tamannotation_scores_rds <- file.path(base_path, "output/q1_pf_characterization/analyses/hc_pf_macrophages_tamannotation_scores.Rds")

pf_macrophages_markergenes_list_rds <- file.path(base_path, "output/q1_pf_characterization/analyses/hc_pf_macrophages_markergenes_list.Rds")
pf_macrophages_markerproteins_list_rds <- file.path(base_path, "output/q1_pf_characterization/analyses/hc_pf_macrophages_markerproteins_list.Rds")

# Plotting parameters
celltype_markers_xlsx <- file.path(base_path, "config/order/celltype_markers.xlsx")
pf_heatmap_order_xlsx <- file.path(base_path, "config/order/q1_pf_heatmap_order.xlsx")
```

### Import data

```{r load subsets}
seurat_pf <- readRDS(seurat_pf_rds)
seurat_pf_myeloid <- readRDS(seurat_pf_myeloid_rds)

seurat_pbmc_pf_liver_colon <- readRDS(seurat_pbmc_pf_liver_colon_rds)
seurat_pf_macrophages <- readRDS(seurat_pf_macrophages_rds)
seurat_liver_macrophages <- readRDS(seurat_liver_macrophages_rds)
seurat_colon_macrophages <- readRDS(seurat_colon_macrophages_rds)
```

```{r load markers}
pf_macrophages_markerproteins_list <- readRDS(pf_macrophages_markerproteins_list_rds)
```

```{r load tam analyses}
tam_markers <- readxl::read_excel(tam_markers_xlsx)
pf_macrophages_tamannotation_scores <- readRDS(pf_macrophages_tamannotation_scores_rds)
```

### Output

```{r figures}
fig1Dir <- file.path(base_path, "docs/manuscript/figures/fig1")
dir.create(fig1Dir, recursive = T)
```

### Plotting parameters

```{r plotting order}
celltype_order_pf_l2_l3 <- readxl::read_excel(celltype_markers_xlsx, col_names = T) %>%
  dplyr::filter(level == "manual_l3",
                modality == "gene") %>%
  dplyr::select(celltype, color) %>%
  unique() %>%
  dplyr::filter(celltype %in% unique(seurat_pf@meta.data[,"manual_l3"])) %>%
  dplyr::mutate(celltype = ifelse(celltype %in% c("Classical monocytes", "Macrophages VCAN+", "Macrophages C1Q+", "Macrophages VCAN+C1Q+", "Macrophages SPP1+"), "Mono-macs", celltype),
                color = ifelse(celltype %in% c("Mono-macs"), "#E31A1C", color)) %>%
  unique() %>%
  dplyr::mutate(number_subset = paste0(1:nrow(.), ". ", celltype))

celltype_order_pf_l4 <- readxl::read_excel(celltype_markers_xlsx, col_names = T) %>%
  dplyr::filter(level == "manual_l4",
                modality == "gene") %>%
  dplyr::select(celltype, color) %>%
  unique() %>%
  dplyr::filter(celltype %in% unique(seurat_pf@meta.data[,"manual_l4"])) %>%
  dplyr::mutate(number_subset = paste0(1:nrow(.), ". ", celltype))
```

```{r colors}
tissue_colors <- c(PF = "#224FBD", PBMC = "#F30000", Colon = "#7B3F00", Liver = "#AA336A")

manual_l2_l3_colors_pf <- celltype_order_pf_l2_l3$color
names(manual_l2_l3_colors_pf) <- celltype_order_pf_l2_l3$celltype

manual_l4_colors_pf <- celltype_order_pf_l4$color
names(manual_l4_colors_pf) <- celltype_order_pf_l4$celltype
```

## Figures

### UMAP HC PF colored by celltype

Here we show the unsupervised clustering analysis using UMAP of PF-derived immune cells annotated using the celltype.

```{r umap hc pf colcelltype, fig.width=7.5, fig.height=7.5}
umap_hc_pf_colcelltype_ggplotobj <- data.frame(CB = colnames(seurat_pf),
                                               Embeddings(seurat_pf[["wnn.umap"]]),
                                               seurat_pf@meta.data) %>%
  dplyr::mutate(celltype = manual_l3,
                celltype = ifelse(celltype %in% c("Classical monocytes", "Macrophages C1Q+", "Macrophages SPP1+", "Macrophages VCAN+", "Macrophages VCAN+C1Q+"), "Mono-macs", celltype),
                celltype = factor(celltype, levels = celltype_order_pf_l2_l3$celltype),
                celltype_number = as.numeric(celltype),
                celltype_w_number = paste0(as.numeric(celltype), ". ", celltype),
                celltype_w_number = factor(celltype_w_number, levels = celltype_order_pf_l2_l3$number_subset)) %>% 
  ggplot(aes(x = wnnUMAP_1, y = wnnUMAP_2)) +
  geom_point_rast(show.legend = T, size = 0.25, aes(col = celltype)) +
  labs(y = "",
       x = "") +
  guides(colour = guide_legend(override.aes = list(size = 3))) +
  scale_color_manual(values = manual_l2_l3_colors_pf) +
  theme_minimal() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        legend.position = "bottom",
        legend.title = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        strip.text.x = element_text(face = "bold"))

pdf(file.path(fig1Dir, "umap_hc_pf_colcelltype.pdf"), width = 7.5, height = 7.5)
print(umap_hc_pf_colcelltype_ggplotobj)
dev.off()
```

### UMAP HC PF Myeloid colored by myeloid GEX markers

Here we characterize the PF-derived ITGAM+ myeloid cells by superimposing the gene expression of the following genes on the respective UMAP configuration:
- Myeloid (ITGAM)
- PDCs (CLEC4C) 
- CDC1s (XCR1)
- CDC2s (CDC1)
- Macrophages (MARCO)
- Macrophages (CD163)
- Monocytes/Monocyte-macrophages (CCR2)
- Classical monocytes/Monocyte-macrophages (CD14)

```{r umap hc pf col_myeloidgexmarkers, fig.width=9, fig.height=9}
umap_hc_pf_myeloid_colmyeloidgenemarkers_ggplotobj_list <- lapply(
  data.frame(CB = colnames(seurat_pf_myeloid),
             Embeddings(seurat_pf_myeloid[["wnn.umap"]]),
             seurat_pf_myeloid@meta.data,
             expr = GetAssayData(seurat_pf_myeloid, assay = "RNA")["CLEC4C",],
             Feature = "CLEC4C") %>%
    dplyr::rows_append(data.frame(CB = colnames(seurat_pf_myeloid),
                                  Embeddings(seurat_pf_myeloid[["wnn.umap"]]),
                                  seurat_pf_myeloid@meta.data,
                                  expr = GetAssayData(seurat_pf_myeloid, assay = "RNA")["XCR1",],
                                  Feature = "XCR1")) %>%
    dplyr::rows_append(data.frame(CB = colnames(seurat_pf_myeloid),
                                  Embeddings(seurat_pf_myeloid[["wnn.umap"]]),
                                  seurat_pf_myeloid@meta.data,
                                  expr = GetAssayData(seurat_pf_myeloid, assay = "RNA")["CD1C",],
                                  Feature = "CD1C")) %>%
    dplyr::rows_append(data.frame(CB = colnames(seurat_pf_myeloid),
                                  Embeddings(seurat_pf_myeloid[["wnn.umap"]]),
                                  seurat_pf_myeloid@meta.data,
                                  expr = GetAssayData(seurat_pf_myeloid, assay = "RNA")["ITGAM",],
                                  Feature = "ITGAM")) %>%
    dplyr::rows_append(data.frame(CB = colnames(seurat_pf_myeloid),
                                  Embeddings(seurat_pf_myeloid[["wnn.umap"]]),
                                  seurat_pf_myeloid@meta.data,
                                  expr = GetAssayData(seurat_pf_myeloid, assay = "RNA")["MARCO",],
                                  Feature = "MARCO")) %>%
    dplyr::rows_append(data.frame(CB = colnames(seurat_pf_myeloid),
                                  Embeddings(seurat_pf_myeloid[["wnn.umap"]]),
                                  seurat_pf_myeloid@meta.data,
                                  expr = GetAssayData(seurat_pf_myeloid, assay = "RNA")["CD14",],
                                  Feature = "CD14")) %>%
    dplyr::rows_append(data.frame(CB = colnames(seurat_pf_myeloid),
                                  Embeddings(seurat_pf_myeloid[["wnn.umap"]]),
                                  seurat_pf_myeloid@meta.data,
                                  expr = GetAssayData(seurat_pf_myeloid, assay = "RNA")["CCR2",],
                                  Feature = "CCR2")) %>%
    dplyr::rows_append(data.frame(CB = colnames(seurat_pf_myeloid),
                                  Embeddings(seurat_pf_myeloid[["wnn.umap"]]),
                                  seurat_pf_myeloid@meta.data,
                                  expr = GetAssayData(seurat_pf_myeloid, assay = "RNA")["CD163",],
                                  Feature = "CD163")) %>%
    dplyr::mutate(expr_rank = rank(expr, ties.method="first"),
                  Feature = factor(Feature, levels = c("CLEC4C", "XCR1", "CD1C", "ITGAM", "MARCO", "CD14", "CCR2", "CD163"))) %>%
    dplyr::filter(manual_l2 %in% c("Macrophages", "CDCs", "PDCs")) %>%
    split(., .$Feature), FUN = function(plotdf){
      plotdf %>% dplyr::arrange(expr) %>%
        ggplot(aes(x = wnnUMAP_1, y = wnnUMAP_2)) +
        geom_point_rast(show.legend = F, size = 0.25, col = "#d3d3d3") +
        geom_point_rast(data = . %>%
                          dplyr::filter(expr>0), aes(col = expr), show.legend = T) +
        guides(colour = guide_legend(override.aes = list(size = 3))) +
        facet_wrap(~Feature) +
        guides(colour = guide_legend(override.aes = list(size = 3),
                                     title = "Expression")) +
        xlim(-6,11) +
        ylim(-5,8) +
        scale_color_viridis() +
        theme_bw() +
        theme(panel.grid.major = element_blank(), 
              panel.grid.minor = element_blank(),
              legend.position = "none",
              axis.title.x = element_blank(),
              axis.title.y = element_blank(),
              strip.text.x = element_text(face = "bold"))
    }
)

pdf(file.path(fig1Dir, "umap_hc_pf_myeloid_colmyeloidgenemarkers.pdf"), width = 9, height = 9)
print(ggarrange(plotlist = umap_hc_pf_myeloid_colmyeloidgenemarkers_ggplotobj_list, nrow = 3, ncol = 3))
dev.off()
```

### UMAP HC PF Myeloid colored by myeloid CITE markers

Here we characterize the PF-derived ITGAM+ myeloid cells by superimposing the protein expression of the following genes on the respective UMAP configuration:
- CDCs (CD1C)
- Macrophages (CD163)
- Monocytes/Monocyte-macrophages (CD192 [CCR2])
- Classical monocytes/Monocyte-macrophages (CD14)

```{r umap hc pf col_myeloidcitemarkers, fig.width=12, fig.height=3}
umap_hc_pf_myeloid_colmyeloidcitemarkers_ggplotobj_list <- lapply(
  data.frame(CB = colnames(seurat_pf_myeloid),
             Embeddings(seurat_pf_myeloid[["wnn.umap"]]),
             seurat_pf_myeloid@meta.data,
             expr = GetAssayData(seurat_pf_myeloid, assay = "CITE")["Hu.CD1c",],
             Feature = "CD1C") %>%
    dplyr::rows_append(data.frame(CB = colnames(seurat_pf_myeloid),
                                  Embeddings(seurat_pf_myeloid[["wnn.umap"]]),
                                  seurat_pf_myeloid@meta.data,
                                  expr = GetAssayData(seurat_pf_myeloid, assay = "CITE")["Hu.CD14-M5E2",],
                                  Feature = "CD14")) %>%
    dplyr::rows_append(data.frame(CB = colnames(seurat_pf_myeloid),
                                  Embeddings(seurat_pf_myeloid[["wnn.umap"]]),
                                  seurat_pf_myeloid@meta.data,
                                  expr = GetAssayData(seurat_pf_myeloid, assay = "CITE")["Hu.CD192",],
                                  Feature = "CD192")) %>%
    dplyr::rows_append(data.frame(CB = colnames(seurat_pf_myeloid),
                                  Embeddings(seurat_pf_myeloid[["wnn.umap"]]),
                                  seurat_pf_myeloid@meta.data,
                                  expr = GetAssayData(seurat_pf_myeloid, assay = "CITE")["Hu.CD163",],
                                  Feature = "CD163")) %>%
    dplyr::mutate(expr_rank = rank(expr, ties.method="first"),
                  Feature = factor(Feature, levels = c("CD1C", "CD14", "CD192", "CD163"))) %>%
    dplyr::filter(manual_l2 %in% c("Macrophages", "CDCs", "PDCs")) %>%
    split(., .$Feature), FUN = function(plotdf){
      plotdf %>% 
        dplyr::arrange(expr) %>%
        ggplot(aes(x = wnnUMAP_1, y = wnnUMAP_2)) +
        geom_point_rast(show.legend = F, size = 0.25, col = "#d3d3d3") +
        geom_point_rast(data = . %>%
                          dplyr::filter(expr>0), aes(col = expr), show.legend = T) +
        guides(colour = guide_legend(override.aes = list(size = 3))) +
        facet_wrap(~Feature) +
        guides(colour = guide_legend(override.aes = list(size = 3),
                                     title = "Expression")) +
        xlim(-6,11) +
        ylim(-5,8) +
        scale_colour_gradient(low = "#ffffff", high = "#FF0000") +
        theme_bw() +
        theme(panel.grid.major = element_blank(), 
              panel.grid.minor = element_blank(),
              legend.position = "none",
              axis.title.x = element_blank(),
              axis.title.y = element_blank(),
              strip.text.x = element_text(face = "bold"))
    }
)

pdf(file.path(fig1Dir, "umap_hc_pf_myeloid_colmyeloidcitemarkers.pdf"), width = 12, height = 3)
print(ggarrange(plotlist = umap_hc_pf_myeloid_colmyeloidcitemarkers_ggplotobj_list, nrow = 1, ncol = 4))
dev.off()
```


### UMAP HC PF macrophages colored by macrophage GEX markers

Here we characterize the PF-derived CD163+ macrophages by superimposing the gene expression of the following genes on the respective UMAP configuration:
- [tissue residency] (VSIG4)
- Macrophages (CD163)
- Monocytes/Monocyte-macrophages/Macrophages VCAN+ (VCAN)
- Monocytes/Monocyte-macrophages (CCR2)
- Macrophages C1Q+ (C1QA)
- Macrophages SPP1+ (SPP1)

```{r umap hc pf macrophages col_macrophagegexmarkers, fig.width=9, fig.height=6}
umap_hc_pf_macrophages_colmacrophagegexmarkers_ggplotobj_list <- lapply(
  data.frame(CB = colnames(seurat_pf_macrophages),
             Embeddings(seurat_pf_macrophages[["gex.umap"]]),
             seurat_pf_macrophages@meta.data,
             expr = GetAssayData(seurat_pf_macrophages, assay = "RNA")["VSIG4",],
             Feature = "VSIG4") %>%
    dplyr::rows_append(data.frame(CB = colnames(seurat_pf_macrophages),
                                  Embeddings(seurat_pf_macrophages[["gex.umap"]]),
                                  seurat_pf_macrophages@meta.data,
                                  expr = GetAssayData(seurat_pf_macrophages, assay = "RNA")["CD163",],
                                  Feature = "CD163")) %>%
    dplyr::rows_append(data.frame(CB = colnames(seurat_pf_macrophages),
                                  Embeddings(seurat_pf_macrophages[["gex.umap"]]),
                                  seurat_pf_macrophages@meta.data,
                                  expr = GetAssayData(seurat_pf_macrophages, assay = "RNA")["VCAN",],
                                  Feature = "VCAN")) %>%
    dplyr::rows_append(data.frame(CB = colnames(seurat_pf_macrophages),
                                  Embeddings(seurat_pf_macrophages[["gex.umap"]]),
                                  seurat_pf_macrophages@meta.data,
                                  expr = GetAssayData(seurat_pf_macrophages, assay = "RNA")["CCR2",],
                                  Feature = "CCR2")) %>%
    dplyr::rows_append(data.frame(CB = colnames(seurat_pf_macrophages),
                                  Embeddings(seurat_pf_macrophages[["gex.umap"]]),
                                  seurat_pf_macrophages@meta.data,
                                  expr = GetAssayData(seurat_pf_macrophages, assay = "RNA")["C1QA",],
                                  Feature = "C1QA")) %>%
    dplyr::rows_append(data.frame(CB = colnames(seurat_pf_macrophages),
                                  Embeddings(seurat_pf_macrophages[["gex.umap"]]),
                                  seurat_pf_macrophages@meta.data,
                                  expr = GetAssayData(seurat_pf_macrophages, assay = "RNA")["SPP1",],
                                  Feature = "SPP1")) %>%
    dplyr::mutate(expr_rank = rank(expr, ties.method="first"),
                  Feature = factor(Feature, levels = c("VSIG4", "CD163", "VCAN", "CCR2", "C1QA", "SPP1"))) %>%
    split(., .$Feature), FUN = function(plotdf){
      plotdf %>% dplyr::arrange(expr) %>%
        ggplot(aes(x = gexUMAP_1, y = gexUMAP_2)) +
        geom_point_rast(show.legend = F, size = 0.25, col = "#d3d3d3") +
        geom_point_rast(data = . %>%
                          dplyr::filter(expr>0), aes(col = expr), show.legend = T) +
        guides(colour = guide_legend(override.aes = list(size = 3))) +
        facet_wrap(~Feature) +
        guides(colour = guide_legend(override.aes = list(size = 3),
                                     title = "Expression")) +
        scale_color_viridis() +
        theme_bw() +
        theme(panel.grid.major = element_blank(), 
              panel.grid.minor = element_blank(),
              legend.position = "none",
              axis.title.x = element_blank(),
              axis.title.y = element_blank(),
              strip.text.x = element_text(face = "bold"))
    }
)

pdf(file.path(fig1Dir, "umap_hc_pf_macrophages_colmacrophagegexmarkers.pdf"), width = 9, height = 6)
print(ggarrange(plotlist = umap_hc_pf_macrophages_colmacrophagegexmarkers_ggplotobj_list, nrow = 2, ncol = 3))
dev.off()
```

### UMAP HC PF macrophages colored by macrophage CITE markers

Here we characterize the CD163+ macrophages by superimposing the protein expression of the following genes on the respective UMAP configuration:
- Macrophages (CD163)

```{r umap hc pf macrophages col_macrophagecitemarkers, fig.width=3, fig.height=3}
umap_hc_pf_macrophages_colmacrophagecitemarkers_ggplotobj <- 
  data.frame(CB = colnames(seurat_pf_macrophages),
             Embeddings(seurat_pf_macrophages[["gex.umap"]]),
             seurat_pf_macrophages@meta.data,
             expr = GetAssayData(seurat_pf_macrophages, assay = "CITE")["Hu.CD163",],
             Feature = "CD163") %>%
  dplyr::mutate(expr_rank = rank(expr, ties.method="first"),
                Feature = factor(Feature, levels = c("CD163"))) %>%
  dplyr::arrange(expr) %>%
  ggplot(aes(x = gexUMAP_1, y = gexUMAP_2)) +
  geom_point_rast(show.legend = F, size = 0.25, col = "#d3d3d3") +
  geom_point_rast(data = . %>%
                    dplyr::filter(expr>0), aes(col = expr), show.legend = T) +
  guides(colour = guide_legend(override.aes = list(size = 3))) +
  facet_wrap(~Feature) +
  guides(colour = guide_legend(override.aes = list(size = 3),
                               title = "Expression")) +
  scale_colour_gradient(low = "#ffffff", high = "#FF0000") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        legend.position = "none",
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        strip.text.x = element_text(face = "bold"))

pdf(file.path(fig1Dir, "umap_hc_pf_macrophages_colmacrophagecitemarkers.pdf"), width = 3, height = 3)
print(umap_hc_pf_macrophages_colmacrophagecitemarkers_ggplotobj)
dev.off()
```

### Boxplot HC PF myeloid percentage relative to all myeloid

Here we prepare the data for scattered boxplots depicting the PF-derived myeloid subsets relative to the total PF-derived myeloid population. Actual data was plotted in Graphpad Prism.

```{r hc pf myeloid l3rl1 data}
seurat_pf_myeloid@meta.data %>%
  dplyr::group_by(manual_l3, SampleID, .drop = T) %>% 
  dplyr::summarise(Ncells = n()) %>%
  dplyr::ungroup() %>%
  tidyr::complete(tidyr::nesting(SampleID), tidyr::nesting(manual_l3), fill = list(Ncells = 0)) %>%
  dplyr::group_by(SampleID) %>%
  dplyr::mutate(Nlrefsample = sum(Ncells),
                Ncellprop = Ncells/Nlrefsample,
                Ncellprop = ifelse(is.na(Ncellprop), 0, Ncellprop),
                Ncellperc = Ncellprop*100,
                Donor = gsub("^(pt[0-9]{2})_.+$", "\\1", SampleID)) %>%
  readr::write_csv(file.path(fig1Dir, "hc_pf_mnp_l3rl1.csv"))
```

### Radarplot HC PF macrophages median TAM classification

Here we present a radar plot representing the median TAM-classification score obtained through UCell for the PF-derived macrophage subsets.

```{r radarplot hc pf macrophages median tam classification, fig.width = 6, fig.height = 5}
radarplot_hc_pf_macrophages_mediantamclassification_ggplotobj <- pf_macrophages_tamannotation_scores %>%
  dplyr::select(c("CellID", "SampleID", "Donor", "manual_l1", "manual_l2", "manual_l3", "manual_l4", "Angio", "IFN", "INFLAM", "LA", "Prolif", "Reg", "RTM")) %>%
  tidyr::pivot_longer(-c("CellID", "SampleID", "Donor", "manual_l1", "manual_l2", "manual_l3", "manual_l4"), names_to = "TAM", values_to = "UCell") %>%
  dplyr::group_by(manual_l4, TAM) %>%
  dplyr::summarize(median_UCell = median(UCell)) %>%
  dplyr::mutate(TAM = factor(TAM, levels = c("Angio", "IFN", "INFLAM", "LA", "Prolif", "Reg", "RTM"))) %>%
  ggplot(aes(x = TAM, y = median_UCell, col = manual_l4, group = manual_l4)) +
  geom_point() +
  geom_encircle(expand = 0, s_shape=1, size = 2) +
  scale_color_manual(values = manual_l4_colors_pf) +
  labs(y = "Median UCell Score") +
  coord_polar() +
  theme_bw() +
  theme(legend.title = element_blank(),
        axis.title.x = element_blank())

pdf(file.path(fig1Dir, "radarplot_hc_pf_macrophages_mediantamclassification.pdf"), width = 6, height=5)
print(radarplot_hc_pf_macrophages_mediantamclassification_ggplotobj)
dev.off()
```

### Dotplot GEX HC PF macrophages M1/M2 markers

Here we present a dotplot representing the percentage expressing cells and median expression of M1 and M2 macrophage marker genes for the PF-derived macrophage subsets.

```{r dotplot gex hc pf col_m1m2gexmarkers, fig.width = 1.7, fig.height=4.25}
m1m2gexmarkers <- c("IL1B", "S100A8", "IDO1", "APOE", "FOLR2", "MRC1", "TREM2", "MERTK")
macrophages_m1m2gexmarkers_gex <- GetAssayData(seurat_pf_macrophages, assay = "RNA")[which(rownames(GetAssayData(seurat_pf_macrophages, assay = "RNA")) %in% m1m2gexmarkers), ]

dotplot_hc_pf_macrophages_m1m2gexmarkers_ggplotobj <- data.frame(FeatureID = rownames(macrophages_m1m2gexmarkers_gex), macrophages_m1m2gexmarkers_gex) %>%
  tidyr::pivot_longer(-c(FeatureID), names_to = "CB", values_to = "nUMIs") %>%
  dplyr::mutate(CB = stringr::str_replace(CB, "\\.", "-")) %>%
  dplyr::inner_join(data.frame(CB = rownames(seurat_pf_macrophages@meta.data), 
                               Celltype = seurat_pf_macrophages@meta.data[,"manual_l4"]), 
                    by = "CB") %>%
  dplyr::mutate(Celltype = factor(Celltype), 
                FeatureID = factor(FeatureID)) %>%
  dplyr::group_by(Celltype, FeatureID, .drop = F) %>%
  dplyr::summarize(Median = median(log1p(nUMIs)),
                   Percentage = mean(nUMIs>0)*100) %>%
  dplyr::mutate(Median = ifelse(is.na(Median), 0, Median),
                Percentage = ifelse(is.na(Percentage), 0, Percentage),
                Celltype = factor(Celltype, c("Macrophages VCAN+", "Macrophages VCAN+C1Q+", "Macrophages C1Q+", "Macrophages SPP1+")),
                FeatureID = factor(FeatureID, rev(m1m2gexmarkers[which(m1m2gexmarkers %in% FeatureID)]))) %>% 
  ggplot(aes(x = Celltype, y = FeatureID, col = Median)) +
  geom_tile(alpha = 0, col = "grey") +
  geom_point(aes(size = Percentage)) +
  theme_bw() +
  theme(legend.position = "bottom", 
        axis.title.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.title.y = element_blank(),
        axis.ticks.y = element_blank(),
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.spacing.x=unit(0, "lines"))

pdf(file.path(fig1Dir, "dotplot_hc_pf_macrophages_m1m2gexmarkers.pdf"), width = 1.7, height = 4.5)
print(dotplot_hc_pf_macrophages_m1m2gexmarkers_ggplotobj)
dev.off()
```

### Heatmap CITE HC PF macrophages markers

Here we present the average expression of the 5 most specifically differentially expressed proteins (obtained through Seurat::FindAllMarkers) for the PF-derived macrophage subsets using a heatmap.

```{r heatmap pf macrophages cite markers, fig.width=2.4, fig.height=4.5}
pf_macrophages_markerproteins_topproteins <- unique(unlist(lapply(pf_macrophages_markerproteins_list[c("Macrophages VCAN+", "Macrophages VCAN+C1Q+", "Macrophages C1Q+","Macrophages SPP1+")], function(i){i[1:5,"gene"]})))
pf_macrophages_markerproteins_topproteins <- pf_macrophages_markerproteins_topproteins[!is.na(pf_macrophages_markerproteins_topproteins)]

seurat_pf_macrophages@meta.data$manual_l4 <- factor(seurat_pf_macrophages@meta.data$manual_l4, levels = c("Macrophages VCAN+", "Macrophages VCAN+C1Q+", "Macrophages C1Q+","Macrophages SPP1+"))

Idents(seurat_pf_macrophages) <- "manual_l4"

marker_pf_macrophages_avepexprl4_pex <- GetAssayData(AverageExpression(seurat_pf_macrophages, return.seurat = T), assay = "CITE")[pf_macrophages_markerproteins_topproteins, c("Macrophages VCAN+", "Macrophages VCAN+C1Q+", "Macrophages C1Q+", "Macrophages SPP1+")] 

heatmap_hc_pf_macrophages_macrophagescitemarkers_complexheatmapobj <- ComplexHeatmap::pheatmap(
  marker_pf_macrophages_avepexprl4_pex, 
  color = PurpleAndYellow(1000),
  labels_row = gsub("Hu\\.", "", pf_macrophages_markerproteins_topproteins),
  cluster_rows = F, 
  cluster_cols = F, 
  scale = "row",
  name = "Scaled median")

pdf(file.path(fig1Dir, "heatmap_hc_pf_macrophages_macrophagescitemarkers.pdf"), width = 2.6, height = 6)
print(heatmap_hc_pf_macrophages_macrophagescitemarkers_complexheatmapobj)
dev.off()
```

### UMAP HC PF, liver, and colon macrophages colored by macrophage GEX markers

Here we characterize the PF-, liver-, and colon-derived CD163+ macrophages by superimposing the gene expression of the following genes on the respective UMAP configuration:
- Macrophages SPP1+ (SPP1)
- Macrophages C1Q+ (C1QA)

```{r umap hc pf liver colon macrophages col_macrophagemarkers, fig.width = 9, fig.height = 6}
umap_hc_pf_liver_colon_macrophages_colmacrophagemarkers_ggplotobj_list <- lapply(
  data.frame(CB = colnames(seurat_pf_macrophages),
             Embeddings(seurat_pf_macrophages[["wnn.umap"]]),
             manual_l2 = seurat_pf_macrophages@meta.data$manual_l2,
             manual_l3 = seurat_pf_macrophages@meta.data$manual_l3,
             expr = GetAssayData(seurat_pf_macrophages, assay = "RNA")["SPP1",],
             Feature = "SPP1",
             Tissue = "PF") %>%
    dplyr::rename(UMAP_1 = wnnUMAP_1,
                  UMAP_2 = wnnUMAP_2) %>%
    dplyr::rows_append(data.frame(CB = colnames(seurat_colon_macrophages),
                                  Embeddings(seurat_colon_macrophages[["umap"]]),
                                  manual_l2 = seurat_colon_macrophages@meta.data$manual_l2,
                                  manual_l3 = seurat_colon_macrophages@meta.data$manual_l3,
                                  expr = GetAssayData(seurat_colon_macrophages, assay = "RNA")["SPP1",],
                                  Feature = "SPP1",
                                  Tissue = "Colon")) %>%
    dplyr::rows_append(data.frame(CB = colnames(seurat_liver_macrophages),
                                  Embeddings(seurat_liver_macrophages[["umap"]]),
                                  manual_l2 = seurat_liver_macrophages@meta.data$manual_l2,
                                  manual_l3 = seurat_liver_macrophages@meta.data$manual_l3,
                                  expr = 0, #Hotfix, else I get an error, there is no SPP1 in liver
                                  Feature = "SPP1",
                                  Tissue = "Liver")) %>%
    dplyr::rows_append(data.frame(CB = colnames(seurat_pf_macrophages),
                                  Embeddings(seurat_pf_macrophages[["wnn.umap"]]),
                                  manual_l2 = seurat_pf_macrophages@meta.data$manual_l2,
                                  manual_l3 = seurat_pf_macrophages@meta.data$manual_l3,
                                  expr = GetAssayData(seurat_pf_macrophages, assay = "RNA")["C1QA",],
                                  Feature = "C1QA",
                                  Tissue = "PF") %>%
                         dplyr::rename(UMAP_1 = wnnUMAP_1,
                                       UMAP_2 = wnnUMAP_2)) %>%
    dplyr::rows_append(data.frame(CB = colnames(seurat_colon_macrophages),
                                  Embeddings(seurat_colon_macrophages[["umap"]]),
                                  manual_l2 = seurat_colon_macrophages@meta.data$manual_l2,
                                  manual_l3 = seurat_colon_macrophages@meta.data$manual_l3,
                                  expr = GetAssayData(seurat_colon_macrophages, assay = "RNA")["C1QA",],
                                  Feature = "C1QA",
                                  Tissue = "Colon")) %>%
    dplyr::rows_append(data.frame(CB = colnames(seurat_liver_macrophages),
                                  Embeddings(seurat_liver_macrophages[["umap"]]),
                                  manual_l2 = seurat_liver_macrophages@meta.data$manual_l2,
                                  manual_l3 = seurat_liver_macrophages@meta.data$manual_l3,
                                  expr = GetAssayData(seurat_liver_macrophages, assay = "RNA")["C1QA",],
                                  Feature = "C1QA",
                                  Tissue = "Liver")) %>%
    split(., list(.$Tissue, .$Feature)), FUN = function(plotdf){
      plotdf %>%
        dplyr::arrange(expr) %>%
        ggplot(aes(x = UMAP_1, y = UMAP_2)) +
        geom_point_rast(show.legend = F, size = 0.25, col = "#d3d3d3") +
        geom_point_rast(data = . %>%
                          dplyr::filter(expr>0), aes(col = expr), show.legend = T) +
        facet_grid(Feature~Tissue) +
        guides(colour = guide_legend(override.aes = list(size = 3),
                                     title = "Expression")) +
        scale_color_viridis() +
        theme_bw() +
        theme(panel.grid.major = element_blank(), 
              panel.grid.minor = element_blank(),
              legend.position = "none",
              axis.text.x = element_blank(),
              axis.ticks.x = element_blank(),
              axis.title.x = element_blank(),
              axis.text.y = element_blank(),
              axis.ticks.y = element_blank(),
              axis.title.y = element_blank(),
              strip.text.x = element_text(face = "bold"))
    }
)

pdf(file.path(fig1Dir, "umap_hc_pf_liver_colon_macrophages_macrophagemarkers.pdf"), width = 9, height = 6)
print(ggarrange(plotlist = umap_hc_pf_liver_colon_macrophages_colmacrophagemarkers_ggplotobj_list, nrow = 2, ncol = 3))
dev.off()
```

### Boxplot PF, Liver, and Colon l2rl0 colored by tissue

Here we prepare the data for scattered boxplots depicting the PF-, colon-, and liver-derived l2 celltypes relative to the total immune cell population. Actual data was plotted in Graphpad Prism.

```{r hc pf liver colon l2rl0}
seurat_pbmc_pf_liver_colon@meta.data %>%
  dplyr::filter(Tissue %in% c("PF", "Colon", "Liver")) %>%
  dplyr::group_by(manual_l2, Donor, Tissue, .drop = T) %>%
  dplyr::summarise(Ncells = n()) %>%
  dplyr::ungroup() %>%
  tidyr::complete(tidyr::nesting(Tissue, Donor), tidyr::nesting(manual_l2), fill = list(Ncells = 0)) %>%
  dplyr::group_by(Donor) %>%
  dplyr::mutate(Nlrefsample = sum(Ncells),
                Ncellprop = Ncells/Nlrefsample,
                Ncellprop = ifelse(is.na(Ncellprop), 0, Ncellprop),
                Ncellperc = Ncellprop*100,
                Donor = gsub("^(pt[0-9]{2})_.+$", "\\1", Donor)) %>%
  readr::write_csv(file = file.path(fig1Dir, "hc_pf_liver_colon_l2rl0.csv"))
```

### Boxplot PF, Liver, and Colon l3rl0 colored by tissue

Here we prepare the data for scattered boxplots depicting the PF-, colon-, and liver-derived l3 celltypes relative to the total immune cell population. Actual data was plotted in Graphpad Prism.

```{r hc pf liver colon l3rl0}
seurat_pbmc_pf_liver_colon@meta.data %>%
  dplyr::filter(Tissue %in% c("PF", "Colon", "Liver")) %>%
  dplyr::group_by(manual_l3, Donor, Tissue, .drop = T) %>%
  dplyr::summarise(Ncells = n()) %>%
  dplyr::ungroup() %>%
  tidyr::complete(tidyr::nesting(Tissue, Donor), tidyr::nesting(manual_l3), fill = list(Ncells = 0)) %>%
  dplyr::group_by(Donor) %>%
  dplyr::mutate(Nlrefsample = sum(Ncells),
                Ncellprop = Ncells/Nlrefsample,
                Ncellprop = ifelse(is.na(Ncellprop), 0, Ncellprop),
                Ncellperc = Ncellprop*100,
                Donor = gsub("^(pt[0-9]{2})_.+$", "\\1", Donor)) %>%
  readr::write_csv(file = file.path(fig1Dir, "hc_pf_liver_colon_l3rl0.csv"))
```
